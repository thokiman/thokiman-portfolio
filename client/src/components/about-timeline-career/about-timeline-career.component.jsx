import React from "react";
import { animateScroll as scroll } from "react-scroll";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectCareerPath } from "../../redux/about/about.selectors";
import TimelineCareerHeader from "../timeline-career-header/timeline-career-header.component";
import TimelineCareerItem from "../timeline-career-item/timeline-career-item.component";
import {
  AboutScrollToTheTop,
  AboutTimelineCareerTitle,
  AboutTimelineContainer,
} from "./about-timeline-career.styles";
import { selectIsSideBarHidden } from "../../redux/header/header.selectors";

export const AboutTimelineCareer = ({
  careerPath: {
    title,
    items: { beforeWorking },
  },
  isSideBarHidden,
}) => {
  const scrollToTop = () => {
    scroll.scrollToTop();
  };

  return (
    <AboutTimelineContainer $issidebarhidden={isSideBarHidden}>
      <TimelineCareerHeader />
      <AboutTimelineCareerTitle>{title}</AboutTimelineCareerTitle>
      <TimelineCareerItem beforeWorking={beforeWorking} />
      <AboutScrollToTheTop onClick={scrollToTop} />
    </AboutTimelineContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  careerPath: selectCareerPath,
  isSideBarHidden: selectIsSideBarHidden,
});

export default connect(mapStateToProps)(AboutTimelineCareer);
