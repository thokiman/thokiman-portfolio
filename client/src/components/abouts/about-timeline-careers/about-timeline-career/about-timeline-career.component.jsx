import React from 'react';
import { connect } from 'react-redux';
import { animateScroll as scroll } from 'react-scroll';
import { createStructuredSelector } from 'reselect';
import {
  selectCareerPath,
  selectLowWaveImage,
} from '../../../../redux/about/about.selectors';
import { selectIsSideBarActive } from '../../../../redux/header/header.selectors';
import TimelineCareerHeader from '../timeline-career-header/timeline-career-header.component';
import TimelineCareerItem from '../timeline-career-item/timeline-career-item.component';
import {
  AboutScrollToTheTop,
  AboutTimelineCareerTitle,
  AboutTimelineContainer,
} from './about-timeline-career.styles';

export const AboutTimelineCareer = ({
  careerPath: {
    title,
    items: { beforeWorking },
  },
  isSideBarActive,
  lowWaveImage,
}) => {
  const scrollToTop = () => {
    scroll.scrollToTop();
  };

  return (
    <AboutTimelineContainer
      $issidebaractive={isSideBarActive}
      $lowWaveImage={lowWaveImage}
    >
      <TimelineCareerHeader />
      <AboutTimelineCareerTitle className='about-timeline-career-title'>
        {title}
      </AboutTimelineCareerTitle>
      <TimelineCareerItem beforeWorking={beforeWorking} />
      <AboutScrollToTheTop
        className='about-timeline-career-title'
        onClick={scrollToTop}
      />
    </AboutTimelineContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  careerPath: selectCareerPath,
  isSideBarActive: selectIsSideBarActive,
  lowWaveImage: selectLowWaveImage,
});

export default connect(mapStateToProps)(AboutTimelineCareer);
