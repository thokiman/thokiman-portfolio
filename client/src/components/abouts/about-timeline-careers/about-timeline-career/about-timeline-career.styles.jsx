import { FaArrowCircleUp } from 'react-icons/fa';
import styled, { css, keyframes } from 'styled-components';

const getTimelineImage = ({ $lowWaveImage }) => {
  return $lowWaveImage.item.iconPath;
};
const SideBarActive = css`
  display: none;
`;
const getSideBarActive = (props) => {
  console.log(props);
  if (props.$issidebarhidden) return SideBarActive;
  return null;
};

export const AboutTimelineContainer = styled.div`
  width: 80%;
  font-family: futura-demi;

  @media only screen and (max-width: 800px) {
    width: 100%;
    background-image: url(${getTimelineImage});
    background-color: rgba(116, 198, 212, 1);
    background-size: cover;
    background-blend-mode: multiply;
    ${getSideBarActive}
  }
`;

const aboutTimelineCareerTitle = keyframes`
0% {
                color:  rgba(116,198,212,0.8)
            }

            100% {
                color: rgba(227,225,228,1.0)
            }
`;

export const AboutTimelineCareerTitle = styled.div`
  text-align: left;
  font-size: 350%;
  padding: 0 10px;
  animation: ${aboutTimelineCareerTitle} 9s ease-in-out infinite alternate both;
`;

const aboutScrollToTheTop = keyframes`
        
            0% {
                transform: translateY(45px);
                animation-timing-function: ease-in;
                opacity: 1;
                color:  rgba(18,129,148,0.9);
            }

            24% {
                opacity: 1;
            }

            40% {
                transform: translateY(24px);
                animation-timing-function: ease-in;
            }

            65% {
                transform: translateY(12px);
                animation-timing-function: ease-in;
                color:  rgba(227,225,228,0.5);
            }

            82% {
                transform: translateY(6px);
                animation-timing-function: ease-in;
            }

            93% {
                transform: translateY(4px);
                animation-timing-function: ease-in;
            }

            25%,
            55%,
            75%,
            87% {
                transform: translateY(0px);
                animation-timing-function: ease-out;
            }

            100% {
                transform: translateY(0px);
                animation-timing-function: ease-out;

                color: rgba(63,66,73,0.9);
            }
        

`;

export const AboutScrollToTheTop = styled(FaArrowCircleUp)`
  color: rgba(227, 225, 228, 0.9);
  position: relative;
  left: -40px;
  bottom: 50px;
  width: 30px;
  height: 30px;
  animation: ${aboutScrollToTheTop} 2s both infinite;
  @media only screen and (max-width: 800px) {
    left: 20px;
  }
`;

AboutTimelineContainer.displayName = 'AboutTimelineContainer';
AboutTimelineCareerTitle.displayName = 'AboutTimelineCareerTitle';
AboutScrollToTheTop.displayName = 'AboutScrollToTheTop';
