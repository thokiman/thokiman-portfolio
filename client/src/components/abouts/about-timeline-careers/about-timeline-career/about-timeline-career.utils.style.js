import {
  hasScreenSize1,
  hasScreenSize1_1,
  hasScreenSize1_2,
  hasScreenSize1_3,
  hasScreenSize2,
  hasScreenSize3_1,
  hasScreenSize3_2,
  hasScreenSize3_3,
  hasScreenSize3_4,
  hasScreenSize3_5,
  hasScreenSize4,
  hasScreenSize5,
  hasScreenSizeWindow,
  hasScreenSizeWindow_1,
} from '../../../../utils/common.component.utils';

export const measureTimelineContainerProps = (viewWidth, viewHeight) => {
  const timelineContainerOffset = 15;
  let timelineContainerProps;

  if (hasScreenSizeWindow(viewWidth, viewHeight)) {
    console.log('about-timeline', 1);
    timelineContainerProps = {
      top: 0,
    };
  } else if (hasScreenSizeWindow_1(viewWidth, viewHeight)) {
    console.log('about-timeline', 2);
    timelineContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1_1(viewWidth, viewHeight)) {
    console.log('about-timeline', 3);
    timelineContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1_2(viewWidth, viewHeight)) {
    console.log('about-timeline', 4);
    timelineContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1_3(viewWidth, viewHeight)) {
    console.log('about-timeline', 5);
    timelineContainerProps = {
      top: 0,
    };
  } else if (hasScreenSize1(viewWidth)) {
    console.log('about-timeline', 6);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize2(viewWidth)) {
    console.log('about-timeline', 7);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize3_1(viewWidth, viewHeight)) {
    console.log('about-timeline', 8);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize3_2(viewWidth, viewHeight)) {
    console.log('about-timeline', 9);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize3_3(viewWidth, viewHeight)) {
    console.log('about-timeline', 10);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize3_4(viewWidth, viewHeight)) {
    console.log('about-timeline', 11);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize3_5(viewWidth, viewHeight)) {
    console.log('about-timeline', 12);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize4(viewWidth, viewHeight)) {
    console.log('about-timeline', 14);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else if (hasScreenSize5(viewWidth)) {
    console.log('about-timeline', 15);
    timelineContainerProps = {
      top: 0 + timelineContainerOffset,
    };
  } else {
    console.log('about-timeline', 16);
    timelineContainerProps = {
      top: 25,
    };
  }

  return timelineContainerProps;
};
