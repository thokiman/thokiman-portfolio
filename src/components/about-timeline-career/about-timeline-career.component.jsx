import React from "react";
import { animateScroll as scroll } from "react-scroll";
import { FaArrowCircleUp } from "react-icons/fa";
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { selectCareerPath } from "../../redux/about/about.selectors";
import TimelineCareerHeader from "../timeline-career-header/timeline-career-header.component";
import TimelineCareerItem from "../timeline-career-item/timeline-career-item.component";
import "./about-timeline-career.styles.scss";

const AboutTimelineCareer = ({
  careerPath: {
    title,
    items: { beforeWorking },
  },
}) => {
  const scrollToTop = () => {
    scroll.scrollToTop();
  };

  return (
    <div className="timeline-container">
      <TimelineCareerHeader />
      <div className="timeline-career-title">{title}</div>
      <TimelineCareerItem beforeWorking={beforeWorking} />
      <FaArrowCircleUp className="scroll-to-the-top" onClick={scrollToTop} />
    </div>
  );
};

const mapStateToProps = createStructuredSelector({
  careerPath: selectCareerPath,
});

export default connect(mapStateToProps)(AboutTimelineCareer);
