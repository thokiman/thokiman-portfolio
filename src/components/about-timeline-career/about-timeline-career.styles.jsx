import styled, { keyframes } from "styled-components";
import { FaArrowCircleUp } from "react-icons/fa";

export const AboutTimelineContainer = styled.div`
  width: 80%;
  font-family: futura-demi;
`;

const aboutTimelineCareerTitle = keyframes`
0% {
                color:  rgba(116,198,212,0.8)
            }

            100% {
                color: rgba(227,225,228,1.0)
            }
`;

export const AboutTimelineCareerTitle = styled.div`
  text-align: left;
  font-size: 350%;
  padding: 0 10px;
  animation: ${aboutTimelineCareerTitle} 9s ease-in-out infinite alternate both;
`;

const aboutScrollToTheTop = keyframes`
        
            0% {
                transform: translateY(45px);
                animation-timing-function: ease-in;
                opacity: 1;
                color:  rgba(18,129,148,0.9);
            }

            24% {
                opacity: 1;
            }

            40% {
                transform: translateY(24px);
                animation-timing-function: ease-in;
            }

            65% {
                transform: translateY(12px);
                animation-timing-function: ease-in;
                color:  rgba(227,225,228,0.5);
            }

            82% {
                transform: translateY(6px);
                animation-timing-function: ease-in;
            }

            93% {
                transform: translateY(4px);
                animation-timing-function: ease-in;
            }

            25%,
            55%,
            75%,
            87% {
                transform: translateY(0px);
                animation-timing-function: ease-out;
            }

            100% {
                transform: translateY(0px);
                animation-timing-function: ease-out;

                color: rgba(63,66,73,0.9);
            }
        

`;

export const AboutScrollToTheTop = styled(FaArrowCircleUp)`
  color: rgba(227, 225, 228, 0.9);
  position: relative;
  left: -40px;
  bottom: 50px;
  width: 30px;
  height: 30px;
  animation: ${aboutScrollToTheTop} 2s both infinite;
`;

AboutTimelineContainer.displayName = "AboutTimelineContainer";
AboutTimelineCareerTitle.displayName = "AboutTimelineCareerTitle";
AboutScrollToTheTop.displayName = "AboutScrollToTheTop";
