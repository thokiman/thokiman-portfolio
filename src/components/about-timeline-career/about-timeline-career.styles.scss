.timeline-container {

    width: 70%;
    height: 86%;
    position: relative;
    top: -240px;
    right: 30px;

    font-family: futura-demi;

    .timeline-career-title {
        background-color: rgba($color: #74c6d4, $alpha: 0.2);

        text-align: left;
        font-size: 350%;
        padding: 0 10px;

        animation: color-change-2x 9s ease-in-out infinite alternate both;

        @keyframes color-change-2x {
            0% {
                color: rgba($color: #74c6d4, $alpha: 0.8);
            }

            100% {
                color: rgba($color:#e3e1e4, $alpha: 1);
            }
        }
    }

    .scroll-to-the-top {
        position: relative;
        bottom: 55px;
        left: 500px;
        color: rgba($color: #e3e1e4, $alpha: 0.9);

        width: 30px;
        height: 30px;
        animation: bounce-bottom 2s both infinite;

        @keyframes bounce-bottom {
            0% {
                transform: translateY(45px);
                animation-timing-function: ease-in;
                opacity: 1;
                color: rgba($color: #128194, $alpha: 0.9);
            }

            24% {
                opacity: 1;
            }

            40% {
                transform: translateY(24px);
                animation-timing-function: ease-in;
            }

            65% {
                transform: translateY(12px);
                animation-timing-function: ease-in;
                color: rgba($color: #e3e1e4, $alpha: 0.5);
            }

            82% {
                transform: translateY(6px);
                animation-timing-function: ease-in;
            }

            93% {
                transform: translateY(4px);
                animation-timing-function: ease-in;
            }

            25%,
            55%,
            75%,
            87% {
                transform: translateY(0px);
                animation-timing-function: ease-out;
            }

            100% {
                transform: translateY(0px);
                animation-timing-function: ease-out;

                color: rgba($color: #3f4249, $alpha: 0.9);
            }
        }
    }
}