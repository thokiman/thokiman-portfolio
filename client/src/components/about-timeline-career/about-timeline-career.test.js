import React from "react";
import { shallow } from "enzyme";

import { AboutTimelineCareer } from "./about-timeline-career.component";

describe("AboutTimelineCareer component", () => {
  let wrapper;
  const mockArray = [];
  const mockText = "";
  const mockFalseBool = false;

  beforeEach(() => {
    const mockProps = {
      careerPath: {
        title: mockText,
        items: { beforeWorking: mockArray },
      },
      isSideBarHidden: mockFalseBool,
    };
    wrapper = shallow(<AboutTimelineCareer {...mockProps} />);
  });

  it("should render AboutTimelineCareer component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
