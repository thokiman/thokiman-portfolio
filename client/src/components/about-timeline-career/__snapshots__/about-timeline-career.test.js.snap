// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutTimelineCareer component should render AboutTimelineCareer component 1`] = `ShallowWrapper {}`;
